src/main/kotlin/com/avail/descriptor/methods/A_Styler.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * A_Styler.kt
+ * Copyright © 1993-2021, The Avail Foundation, LLC.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of the contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.avail.descriptor.methods
+
+import com.avail.descriptor.functions.A_Function
+import com.avail.descriptor.module.A_Module
+import com.avail.descriptor.representation.A_BasicObject
+import com.avail.descriptor.representation.A_BasicObject.Companion.dispatch
+import com.avail.descriptor.representation.AvailObject
+
+/**
+ * `A_Styler` is an interface that specifies the operations specific to
+ * [stylers][StylerDescriptor] of a method [definition][A_Definition] in Avail.
+ * It's a sub-interface of [A_BasicObject], the interface that defines the
+ * behavior that all [AvailObject]s are required to support.
+ *
+ * @author Mark van Gulik &lt;mark@availlang.org&gt;
+ */
+interface A_Styler : A_BasicObject {
+	companion object
+	{
+		/**
+		 * Answer the [A_Function] that implements this styler's logic.
+		 */
+		val A_Styler.function: A_Function get() = dispatch { o_Function(it) }
+
+		/**
+		 * Answer the [method][MethodDescriptor] that this
+		 * [definition][DefinitionDescriptor] is for.
+		 *
+		 * @return
+		 *   The definition's method.
+		 */
+		val A_Styler.definition: A_Definition get() =
```

Comment:
1. Not quite.  The methodName/sourceModule/lineNumber inside the style is where the user should be hyperlink-redirected if they Cmd-click (or something) on the styled text.  So it'll usually take the user to the method/macro definition that the styler is attached to, but the styler function can choose a more useful target instead.  Like if the code says `"foo's banana"`, clicking on `banana` might take you to the clause where `banana` is mentioned in `foo`'s class definition.  Furthermore, the `"_'s⁇banana"` method is generated, so that's about the only reasonable place to go.
2. I'm not sure how `generated` gets populated – that'll be discovered/invented when we start actually creating styles.  The semanticClassifier will be a string that a styler function will fill in.  The default styler function will be somewhat complex, and will produce some pre-built semantic classifier names.  If the most suitable method/macro definition for an invocation is a primitive function, perhaps the default styler can look for a @SemanticClassifier annotation on the primitive class, and extract the string from the annotation.
3. Two parts:
  a. Correct.
  b. If it's a method definition or macro definition, there's a body block that captures the line number and module.  And in (2), I proposed putting the semanticClassifier information in an annotation on the primitive.  If we have an invocation of a non-primitive, the default styler will have to style the tokens by their kind, where keyword tokens get some style and operator tokens get some other style.  Maybe also make a distinction for public/private.  If we want more control, we'll have to attach a styler function to that definition, or introduce some sort of optional "declarations" in the Avail code for the default styler to interpret.  Maybe something like "Method_style_is_", where the second argument is the semanticClassifier name.

This pull request is mostly about adding the (latent) capability to talk about styles and stylers, rather than actually implement stylers.  We'll create a rudimentary  default styler later, then improve it.